clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -187,9 +193,9 @@ protected void onJoinComplete(int generation,
         // give the assignor a chance to update internal state based on the received assignment
         assignor.onAssignment(assignment);
 
-        // restart the autocommit task if needed
+        // reschedule the auto commit starting from now
         if (autoCommitEnabled)
-            autoCommitTask.enable();
+            autoCommitTask.reschedule();
```

Comment:
Wasn't there a good reason for using `enable`/`disable` rather than just `reschedule`? I guess maybe we don't need to since auto commit is best effort anyway so things like rebalance callbacks wont affect any of this?

Would you mind just quickly summarizing how this impacts https://issues.apache.org/jira/browse/KAFKA-2860 which is where a bunch of this code that is being removed was added (all the `enable`/`disable`/`reschedule` code and `requestInFlight`.